canals/pipeline/pipeline.py

```diff
@@ -152,8 +153,18 @@ def from_dict(cls: Type[T], data: Dict[str, Any], **kwargs) -> T:
             else:
                 if "type" not in component_data:
                     raise PipelineError(f"Missing 'type' in component '{name}'")
+
                 if component_data["type"] not in component.registry:
-                    raise PipelineError(f"Component '{component_data['type']}' not imported.")
+                    try:
+                        # Import the module first...
+                        module, _ = component_data["type"].rsplit(".", 1)
+                        importlib.import_module(module)
+                        # ...then try again
+                        if component_data["type"] not in component.registry:
+                            raise PipelineError()
```

Comment:
Nitpick: even if it's close to impossible, I'd add a message to this error asking to open a bug report to Canals.